server/public/model/ldap.go

```diff
@@ -3,8 +3,48 @@
 
 package model
 
+import (
+	"strconv"
+)
+
 const (
 	UserAuthServiceLdap       = "ldap"
 	LdapPublicCertificateName = "ldap-public.crt"
 	LdapPrivateKeyName        = "ldap-private.key"
 )
+
+type LdapSyncOptions struct {
+	ReAddRemovedMembers *bool `json:"re_add_removed_members"`
+}
+
+// ToMap converts a LdapSyncOptions to a map[string]string.
+func (opts *LdapSyncOptions) ToMap() map[string]string {
+	m := map[string]string{}
+	if opts == nil {
+		return m
+	}
+
+	if opts.ReAddRemovedMembers != nil {
+		m["re_add_removed_members"] = strconv.FormatBool(*opts.ReAddRemovedMembers)
+	}
+
+	return m
+}
+
+// FromMap populates a LdapSyncOptions from a map[string]string.
+// It should never be called on a nil pointer.
+func (opts *LdapSyncOptions) FromMap(m map[string]string) {
+	if opts == nil {
+		panic("opts is nil")
```

Comment:
Why panic here? Shouldn't we just return nil?

Reply:
Calling `FromMap` with `nil` options is never a good idea. A method can not change its receiver. The panic makes it obvious that it is not valid.